src/main/java/org/embulk/output/sf_bulk_api/SForceTransactionalPageOutput.java

```diff
@@ -41,23 +51,31 @@ public void add(Page page) {
         records.add(record);
         if (records.size() >= BATCH_SIZE) {
           try {
-            forceClient.action(records);
+            failures += forceClient.action(records);
+            failed = failures != 0;
           } catch (ApiFault e) {
             // even if some records failed to register, processing continues.
-            logger.error(e.getExceptionCode().toString() + ":" + e.getExceptionMessage(), e);
+            failures += errorHandler.handleFault(records, e);
+            failed = true;
           }
           records = new ArrayList<>();
         }
       }
 
       if (CollectionUtils.isNotEmpty(records)) {
         try {
-          forceClient.action(records);
+          failures += forceClient.action(records);
+          failed = failures != 0;
         } catch (ApiFault e) {
-          logger.error(e.getExceptionCode().toString() + ":" + e.getExceptionMessage(), e);
+          failures += errorHandler.handleFault(records, e);
+          failed = true;
         }
       }
+    } catch (AbortException e) {
+      logger.error(e.getMessage(), e);
```

Comment:
Don't we need to do `failed = true` here?
(no problem because it will throws exception and commit is not called?)

```suggestion
      logger.error(e.getMessage(), e);
      failed = true;
```

Reply:
> no problem because it will throws exception and commit is not called?

Yes, if throw exception here, commit is not called (abort is called instead)